qcodes/dataset/data_set_protocol.py

```diff
@@ -414,16 +426,30 @@ def _export_data(
             file_name = self._export_file_name(
                 prefix=prefix, export_type=DataExportType.NETCDF
             )
-            return self._export_as_netcdf(path=path, file_name=file_name)
+            export_path = self._export_as_netcdf(path=path, file_name=file_name)
 
         elif DataExportType.CSV == export_type:
             file_name = self._export_file_name(
                 prefix=prefix, export_type=DataExportType.CSV
             )
-            return self._export_as_csv(path=path, file_name=file_name)
+            export_path = self._export_as_csv(path=path, file_name=file_name)
 
         else:
-            return None
+            export_path = None
+
+        # for unknown reason entrypoints registered in pyproct.toml shows up
+        # twice here
+        export_callbacks = set(entry_points(group="qcodes.dataset.on_export"))
+
+        if len(export_callbacks) == 0:
+            LOG.info("No export callbacks defined")
+
+        for export_callback in export_callbacks:
+            export_callback_function = export_callback.load()
+            LOG.info("Executing on_export callback {export_callback.name}")
+            export_callback_function(export_path)
```

Comment:
This should probably take a Path object and not a str. Also update to reflect that this may be None

Reply:
Do you think it could make sense to wrap this call with a try-except, so that the hook cannot crash the entire app, or does the hook mechanism already do this?

Reply:
Yes thats probably a good idea

Reply:
Its probably also a good idea to wrap the logic into a function that returns a collection of functions to execute. That way we can mock that in a test so that we can test that functions are exectuted regardless of the entrypoint logic.